### PR TITLE
[test] Add s390x support for test/IRGen/big_types_corner_cases.swift

### PR DESCRIPTION
**Pull request by levivic**
Add s390x support for test/IRGen/big_types_corner_cases.swift

### CODE REVIEW

File: test/IRGen/big_types_corner_cases.swift [hunk 1]
```diff
@@ -335,7 +335,7 @@ public extension sr8076_QueryHandler {
     }
 
 // CHECK-LABEL-64: define{{( dllexport)?}}{{( protected)?}} swiftcc { i64, i64 } @"$s22big_types_corner_cases19sr8076_QueryHandlerPAAE13forceHandle_45query8ReturnedQyd___SbAA9BigStructVcSgtqd___tKAA0e1_F0Rd__lF"(%swift.opaque* noalias nocapture, %swift.opaque* noalias nocapture, %swift.type*{{.*}}, %swift.type*{{.*}}, i8** {{.*}}.sr8076_QueryHandler, i8** {{.*}}.sr8076_Query, %swift.opaque* noalias nocapture swiftself, %swift.error** swifterror)
-// CHECK-64: {{.*}} = call swiftcc { i64, i64 } {{.*}}(%swift.opaque* noalias nocapture {{.*}}, %swift.opaque* noalias nocapture {{.*}}, %swift.refcounted* swiftself {{.*}}, %swift.error** noalias nocapture swifterror {{.*}})
+// CHECK-64: {{.*}} = call swiftcc { i64, i64 } {{.*}}(%swift.opaque* noalias nocapture {{.*}}, %swift.opaque* noalias nocapture {{.*}}, %swift.refcounted* swiftself {{.*}}, %swift.error** noalias nocapture{{( swifterror)?}}{{( dereferenceable\(\{\{.\}\}\))?}} {{.*}})
```
Comment (compnerd) [inline test/IRGen/big_types_corner_cases.swift]: Why the addition of the check for `dereferenceable`?  Does S/390 not support the swift calling convention?

Comment (levivic) [reply test/IRGen/big_types_corner_cases.swift]: Swift calling convention is already supported on s390x https://reviews.llvm.org/rL267823. However,`swifterror` is replaced by `dereferenceable({{.}})` on s390x.

Comment (compnerd) [reply test/IRGen/big_types_corner_cases.swift]: That sounds wrong.  The parameter should be a `swifterror` type.  `dereferenceable` would be applicable to all the targets.  This sounds like either a bug in clang or LLVM for the s390 target.

Comment (levivic) [reply test/IRGen/big_types_corner_cases.swift]: agree, any suggestions on how to narrow down such bug?

Comment (compnerd) [reply test/IRGen/big_types_corner_cases.swift]: I would look at the output of `-emit-silgen` from the `swiftc` frontend.

**Comment by CodaFi** [comment]
@levivic Let's get this moving again.  Have you had time to look into this issue?

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by CodaFi** [comment]
Alright, I'm going to close this out due to age and inactivity. I think this change is still good to have for big-endian targets but I don't have time to shepherd this through. Please consider reopening this pull request when you have the time.

**Pull request closed without merging**